src/netutil.rs

```diff
@@ -133,9 +146,79 @@ pub fn bind_to(port: u16, reuseaddr: bool) -> io::Result<UdpSocket> {
     }
 }
 
+#[cfg(not(target_os = "linux"))]
+pub fn recv_mmsg(socket: &UdpSocket, packets: &mut [Packet]) -> io::Result<usize> {
+    let mut i = 0;
+    socket.set_nonblocking(false)?;
+    for n in 0..NUM_RCVMMSGS {
```

Comment:
you can use packets.len() here, because you can't do any more reading if the slice (which represents a "read with limit" is full)